addons/mail/static/src/model/record.js

```diff
@@ -409,20 +416,75 @@ export class Record {
         return !this.in(collection);
     }
 
-    toData() {
+    /**
+     * Converts the current record and its related data into Store insert-able data.
+     * @param {Array<string> | { depth: boolean }} options Configuration options or an array of field names.
+     * @returns {Object} A data object grouped by model names.
+     */
+    toData(options = { depth: false }) {
+        const prefix = this._getPyModelName();
+        const ongoing = {
+            seenRecords: new Set(),
+            storeData: {},
+            depth: options.depth,
+            fields: undefined,
+        };
+        if (Array.isArray(options)) {
+            ongoing.fields = options.map((field) => `${prefix}.${field}`);
+        }
+        this._toData(ongoing, prefix);
+        return ongoing.storeData;
+    }
+
+    _cleanupData(data) {
+        const fieldsToDelete = [
+            "_",
+            "_fieldsValue",
+            "_proxy",
+            "_proxyInternal",
+            "_raw",
+            "env",
+            "Model",
+        ];
+        fieldsToDelete.forEach((field) => delete data[field]);
+    }
+
+    _getPyModelName() {
+        return this.Model.getName();
+    }
```

Comment:
the name of this function is not really correct as it could return a JS model name too for JS only models